### PR TITLE
Fix: Test data leaking into dev

### PR DESCRIPTION
**Pull request by FadhlanR**
image (1)

This issue happened because we didn't clear the `recent-files` from localStorage after `operator-mode-test` is executed. As we can see in other tests we always clear the `recent-files` after we're executing the tests.

https://github.com/cardstack/boxel/blob/7a4b13e6a51936df4ed52e4273e362004b37dc2a/packages/host/tests/acceptance/code-submode/recent-files-test.ts#L182
https://github.com/cardstack/boxel/blob/7a4b13e6a51936df4ed52e4273e362004b37dc2a/packages/host/tests/acceptance/code-submode/file-tree-test.ts#L199

### CODE REVIEW

**Review by tintinthong — APPROVED** [review]
Nice!

**Review by jurgenwerk — APPROVED** [review]

**Comment by habdelra** [comment]
It’s really strange though that the real local storage was used in the first place. I see we use the mock storage in the test. Can you confirm that we also use `import window` in the place where this value is set?

**Comment by backspace** [comment]
> It’s really strange though that the real local storage was used in the first place. I see we use the mock storage in the test. Can you confirm that we also use `import window` in the place where this value is set?

ya if we use the `setupWindowMock` test helper like in other tests, we shouldn’t need to clear things manually.

**Comment by FadhlanR** [comment]
@habdelra @backspace Thanks for the feedbacks. I had added `setupWindowMock` to the test.

File: packages/host/tests/integration/components/operator-mode-test.gts [hunk 1]
```diff
@@ -81,16 +82,19 @@ module('Integration | operator-mode', function (hooks) {
   );
   setupServerSentEvents(hooks);
   setupMatrixServiceMock(hooks);
+  setupWindowMock(hooks);
   let noop = () => {};
 
   hooks.afterEach(async function () {
     localStorage.removeItem('recent-cards');
+    localStorage.removeItem('recent-files');
```
Comment (backspace) [inline packages/host/tests/integration/components/operator-mode-test.gts]: I think this clears “real” `localStorage`, if you want to use the mocked one in tests you need `import window from 'ember-window-mock'` and then `window.localStorage`

BUT I think none of this is needed anymore, if everything is set up to use the conditional `window` it’s cleaned up automatically and these `removeItem`s are redundant.

**Comment by backspace** [comment]
I opened #1234 to see if I’m correct that we don’t need `window.localStorage.removeItem` in the hooks, I think that clearing the mock `localStorage` comes for free and these are redundant, let’s see if the tests pass

**Comment by backspace** [comment]
> I opened #1234 to see if I’m correct that we don’t need `window.localStorage.removeItem` in the hooks, I think that clearing the mock `localStorage` comes for free and these are redundant, let’s see if the tests pass

It worked! There are some lint errors as I failed to remove the `window` import. Feel free to merge in that PR if you want, or close it if not.

**Comment by FadhlanR** [comment]
> > I opened #1234 to see if I’m correct that we don’t need `window.localStorage.removeItem` in the hooks, I think that clearing the mock `localStorage` comes for free and these are redundant, let’s see if the tests pass
> 
> It worked! There are some lint errors as I failed to remove the `window` import. Feel free to merge in that PR if you want, or close it if not.

Great! I'll merge it Thanks!

File: packages/host/app/components/ai-assistant/panel.gts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { tracked, cached } from '@glimmer/tracking';
 
 import { restartableTask, timeout } from 'ember-concurrency';
 import { Velcro } from 'ember-velcro';
+import window from 'ember-window-mock';
```
Comment (burieberry) [inline packages/host/app/components/ai-assistant/panel.gts:12]: is it ok to use the mock window here in a non-test file?

File: packages/host/tests/integration/components/operator-mode-test.gts [hunk 3]
```diff
@@ -13,6 +13,8 @@ import {
 import GlimmerComponent from '@glimmer/component';
 
 import { setupRenderingTest } from 'ember-qunit';
+import window from 'ember-window-mock';
+import { setupWindowMock } from 'ember-window-mock/test-support';
```
Comment (burieberry) [inline packages/host/tests/integration/components/operator-mode-test.gts:17]: 👍

File: packages/host/app/components/ai-assistant/panel.gts (hunk 2 above)
Comment (habdelra) [reply packages/host/app/components/ai-assistant/panel.gts:12]: yes! this is precisely why we run into these problems. We _always_ need to be using the mock window anywhere in the codebase that we want to use the window object to access local storage. You need to use mock window on both sides: the place that you set the value in local storage and the place that you read the value from local storage.

**Pull request merged**